### PR TITLE
Avoid hitting slow path every time for aliased method

### PR DESCRIPTION
**Pull request by djudd-stripe**
In the edge case where we've called `alias` or `alias_method` after adding a `sig` to a method, we previously would hit the initial, slow `replace_method` wrapper every time - even if we used `checked(:never)` on the `sig`.

This avoids that by using `__callee__` plus `Method#original_name` to find the `Signature` object for the original method, if we've hit the alias, and then using that to generate a fast path more or less as we would have on the original method.

### Motivation
Least surprise: if `alias_method` isn't a performance problem for vanilla Ruby, Sorbet shouldn't make it one. We use it a couple of times in T::Props ourselves!

### Test plan
Expanded aliasing tests, also ran benchmarks and saw no notable change. (We don't bench this edge case, which seems fine.)

### CODE REVIEW

File: gems/sorbet-runtime/lib/types/private/methods/_methods.rb [hunk 1]
```diff
@@ -252,6 +248,43 @@ def self._on_method_added(hook_mod, method_name, is_singleton_method: false)
...
+        T::Private::Methods.unwrap_method(mod, method_sig, new_method)
+      else
+        # Unchecked, so just make `alias_method` behave as if it had been called pre-sig.
+        mod.send(:alias_method, method_name, original_name)
+      end
+    else
+      msg = "`sig` not present for method `#{method_name}` but you're trying to run it anyways. " \
+        "This should only be executed if you used `alias_method` to grab a handle to a method after `sig`ing it, but that clearly isn't what you are doing. " \
+        "Maybe look to see if an exception was thrown in your `sig` lambda or somehow else your `sig` wasn't actually applied to the method. " \
+        "Contact #dev-productivity if you're really stuck."
```
Comment (jez) [inline gems/sorbet-runtime/lib/types/private/methods/_methods.rb:271]: Is this messages still relevant?

Comment (djudd-stripe) [reply gems/sorbet-runtime/lib/types/private/methods/_methods.rb]: Not really, I think!

Also I removed the soft-assert fallback because it was broken (I'd stopped threading through the return value) and without any way to cover that with tests it didn't seem worth maintaining.
